### PR TITLE
[MM-42522] - The reply-to email address in our Cloud trial emails is no-reply@mattermost.com instead of feedback-cloud@mattermost.com

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
We need to customize the reply to address for some cloud emails like the welcome email. This PR makes it possible.

#### Ticket Link
[MM-42522](https://mattermost.atlassian.net/browse/MM-42522)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: app/email/utils.go [hunk 1]
```diff
@@ -8,9 +8,14 @@ import (
...
 )
 
-func (es *Service) mailServiceConfig() *mail.SMTPConfig {
+func (es *Service) mailServiceConfig(replyToAddress string) *mail.SMTPConfig {
 	emailSettings := es.config().EmailSettings
 	hostname := utils.GetHostnameFromSiteURL(*es.config().ServiceSettings.SiteURL)
+
+	if replyToAddress == "" {
+		replyToAddress = *emailSettings.ReplyToAddress
+	}
```
Comment (marianunez) [inline app/email/utils.go:17]: If the replyToAddress is part of the config, why can we default this to the address we want for cloud and there would be no need to do a special case override here https://github.com/mattermost/enterprise/blob/6b3accafc357e037105fafe14643396708eb90b9/cloud/config/cloud_defaults.json#L16

**Review by marianunez — CHANGES_REQUESTED** [review]

File: app/email/utils.go (hunk 1 above)
Comment (AGMETEOR) [reply app/email/utils.go:17]: @marianunez I thought so too, but that means all cloud emails reply to would be that value set in the config. But based on this conversation https://mattermost.atlassian.net/browse/MM-42522?focusedCommentId=125238, Jason mentioned that it shouldn't be that way

Comment (marianunez) [reply app/email/utils.go:17]: I see, but if we are overriding it here at the highest level of the Mail Service, wouldn't this change affect all emails anyways?

Comment (marianunez) [reply app/email/utils.go:17]: I see now that we instantiate this for each email. I thought this was a shared config for all emails. 👍

**Review by marianunez — APPROVED** [review]
Can we also do this change for all emails coming from CWS webhook? Trial and payment failure emails

File: app/email/email.go [hunk 2]
```diff
@@ -310,7 +310,7 @@ func (es *Service) SendCloudWelcomeEmail(userEmail, locale, teamInviteID, workSp
 		return err
 	}
 
-	if err := es.sendMail(userEmail, subject, body); err != nil {
+	if err := es.sendEmailWithCustomReplyTo(userEmail, subject, body, *es.config().SupportSettings.SupportEmail); err != nil {
```
Comment (marianunez) [inline app/email/email.go:313]: Any tests we can add to cover this?

**Review by neallred — APPROVED** [review]

File: app/email/email.go (hunk 2 above)
Comment (AGMETEOR) [reply app/email/email.go:315]: Added a unit test

**Pull request merged**